pkg/kubelet/prober/prober.go

```diff
@@ -173,6 +173,12 @@ func (pb *prober) runProbe(probeType probeType, p *v1.Probe, pod *v1.Pod, status
 		if host == "" {
 			host = status.PodIP
 		}
+		var initDelay int32
+		if probeType == readiness {
+			initDelay = p.InitialDelaySeconds
+		}
+		timeout = time.Duration(p.TimeoutSeconds+initDelay) * time.Second
```

Comment:
Delaying the probe instead of increasing the readiness timeout seems more correct to me.

Reply:
And it seems like kubelet does delay the probe https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/prober/worker.go#L226 based on init delay seconds.

Reply:
I saw that code (which has been there since 2016) before making the change.

Without this fix (and without specifying timeoutSeconds), I observed:
```
zyu@zyu-mac t-kube % oc set probe dc/workshop --readiness --get-url=http://:8080/5s_delay --initial-delay-seconds=10
deploymentconfig.apps.openshift.io/workshop probes updated
zyu@zyu-mac t-kube % oc get pods
NAME                READY   STATUS              RESTARTS   AGE
workshop-1-build    0/1     Completed           0          3m40s
workshop-1-deploy   0/1     Completed           0          2m20s
workshop-1-q6pl6    1/1     Running             0          2m14s
workshop-2-7nk6m    0/1     Terminating         0          84s
workshop-3-deploy   0/1     ContainerCreating   0          3s

Then:

zyu@zyu-mac t-kube % oc get pods
NAME                READY   STATUS      RESTARTS   AGE
workshop-1-build    0/1     Completed   0          4m29s
workshop-1-deploy   0/1     Completed   0          3m9s
workshop-1-q6pl6    1/1     Running     0          3m3s
workshop-3-deploy   1/1     Running     0          52s
workshop-3-mrt9x    0/1     Running     0          47s
```